contracts/child/validator/ValidatorSet.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Snapshot.sol";
+import "../../interfaces/child/validator/IValidatorSet.sol";
+import "../../interfaces/IStateSender.sol";
+import "../System.sol";
+
+contract ValidatorSet is IValidatorSet, ERC20Snapshot, System, CVSWithdrawal {
+    using WithdrawalQueueLib for WithdrawalQueue;
+
+    bytes32 private constant STAKE_SIG = keccak256("STAKE");
+    bytes32 private constant UNSTAKE_SIG = keccak256("UNSTAKE");
+    bytes32 private constant SLASH_SIG = keccak256("SLASH");
+
+    IStateSender private immutable STATE_SENDER;
+    address private immutable STATE_RECEIVER;
+    address private immutable ROOT_CHAIN_MANAGER;
+    uint256 private immutable _EPOCH_SIZE;
+
+    event NewEpoch(uint256 indexed id, uint256 indexed startBlock, uint256 indexed endBlock, bytes32 epochRoot);
+
+    constructor(
+        address stateSender,
+        address stateReceiver,
+        address rootChainManager,
+        uint256 epochSize_,
+        ValidatorInit[] memory initalValidators
+    ) ERC20("ValidatorSet", "VSET") {
+        STATE_SENDER = IStateSender(stateSender);
+        STATE_RECEIVER = stateReceiver;
+        ROOT_CHAIN_MANAGER = rootChainManager;
+        _EPOCH_SIZE = epochSize_;
+        for (uint256 i = 0; i < initalValidators.length; i++) {
```

Comment:
Minor gasopt here with unchecked and prefix increment.

Reply:
fixed